plugins/analysis-icu/build.gradle

```diff
@@ -18,17 +18,19 @@ import de.thetaphi.forbiddenapis.gradle.CheckForbiddenApis
  * specific language governing permissions and limitations
  * under the License.
  */
-apply plugin: 'elasticsearch.rest-resources'
+apply plugin: 'elasticsearch.yaml-rest-test'
 
 esplugin {
   description 'The ICU Analysis plugin integrates the Lucene ICU module into Elasticsearch, adding ICU-related analysis components.'
   classname 'org.elasticsearch.plugin.analysis.icu.AnalysisICUPlugin'
 }
 
-tasks.withType(CheckForbiddenApis).configureEach {
-  signatures += [
-    "com.ibm.icu.text.Collator#getInstance() @ Don't use default locale, use getInstance(ULocale) instead"
-  ]
+tasks.withType(CheckForbiddenApis).configureEach { task ->
+  if(task.name.contains("YamlRestTest") == false) {
```

Comment:
this is needed since the yamlRestRest does not have access to that class

Reply:
Maybe this should be explicit, applying the signature to `forbiddanApisMain` and `forbiddenApisTest`, rather than all CheckForbiddenApis tasks? We are likely to continue adding more source sets in the future, so having configuration apply to all source sets would be problematic.

Reply:
:+1: done [`5c75412` (#59444)](https://github.com/elastic/elasticsearch/pull/59444/commits/5c75412f830b1ea938bbfee39b2a20c4bc6f90d8) for the main source set (probably not the extra config noise for the test and internalTestCluster source sets)